server/src/main/java/org/opensearch/gateway/GatewayMetaState.java

```diff
@@ -674,7 +678,22 @@ public void setLastAcceptedState(ClusterState clusterState) {
             try {
                 final ClusterMetadataManifest manifest;
                 if (shouldWriteFullClusterState(clusterState)) {
-                    manifest = remoteClusterStateService.writeFullMetadata(clusterState);
+                    if (clusterState.metadata().clusterUUIDCommitted() == true) {
+                        final Optional<ClusterMetadataManifest> latestManifest = remoteClusterStateService.getLatestClusterMetadataManifest(
+                            clusterState.getClusterName().value(),
+                            clusterState.metadata().clusterUUID()
+                        );
+                        if (latestManifest.isPresent()) {
+                            previousClusterUUID = latestManifest.get().getPreviousClusterUUID();
```

Comment:
Lets add a comment that for current clusterUUID, the previousUUID shouldn't change and hence it fetches from remote.